libs/langchain-google-common/src/media.ts

```diff
@@ -0,0 +1,775 @@
+  method: GoogleAbstractedClientOpsMethod;
+  name: string;
+}
+
+export class AIStudioFileDownloadConnection<
+  ResponseType extends GoogleResponse,
+  AuthOptions
+> extends GoogleDownloadConnection<AsyncCallerCallOptions, ResponseType, AuthOptions> {
+
+  method: GoogleAbstractedClientOpsMethod;
+
+  name: string;
+
+  apiVersion = "v1beta";
+
+  constructor(
+    fields: AIStudioFileDownloadConnectionParams<AuthOptions>,
+    caller: AsyncCaller,
+    client: GoogleAbstractedClient
+  ) {
+    super(fields, caller, client);
+    this.method = fields.method;
+    this.name = fields.name;
+  }
+
+  buildMethod(): GoogleAbstractedClientOpsMethod {
+    return this.method
+  }
+
+  async buildUrl(): Promise<string> {
+    return `https://generativelanguage.googleapis.com/${this.apiVersion}/files/${this.name}`;
+  }
+}
+
+export interface BlobStoreAIStudioFileBaseParams<AuthOptions>
+  extends BlobStoreGoogleParams<AuthOptions> {
+  retryTime?: number;
+}
+
+export abstract class BlobStoreAIStudioFileBase<
+  AuthOptions
+> extends BlobStoreGoogle<AIStudioFileResponse, AuthOptions> {
+
+  params?: BlobStoreAIStudioFileBaseParams<AuthOptions>;
+
+  retryTime: number = 1000;
+
+  constructor(fields?: BlobStoreAIStudioFileBaseParams<AuthOptions>) {
+    const params: BlobStoreAIStudioFileBaseParams<AuthOptions> = {
+      defaultStoreOptions: {
+        pathPrefix: "https://generativelanguage.googleapis.com/v1beta/files/",
+        actionIfInvalid: "removePath",
+      },
+      ...fields,
+    };
+    super(params);
+    this.params = params;
+    this.retryTime = params?.retryTime ?? this.retryTime ?? 1000;
+  }
+
+  _pathToName(path: string): string {
+    return path.split("/").pop() ?? path;
+  }
+
+  abstract buildAbstractedClient(
+    fields?: BlobStoreGoogleParams<AuthOptions>
+  ): GoogleAbstractedClient;
+
+  buildApiKeyClient(apiKey: string): GoogleAbstractedClient {
+    return new ApiKeyGoogleAuth(apiKey);
+  }
+
+  buildApiKey(fields?: BlobStoreGoogleParams<AuthOptions>): string | undefined {
+    return fields?.apiKey ?? getEnvironmentVariable("GOOGLE_API_KEY");
+  }
+
+  buildClient(
+    fields?: BlobStoreGoogleParams<AuthOptions>
+  ): GoogleAbstractedClient {
+    const apiKey = this.buildApiKey(fields);
+    if (apiKey) {
+      return this.buildApiKeyClient(apiKey);
+    } else {
+      // TODO: Test that you can use OAuth to access
+      return this.buildAbstractedClient(fields);
+    }
+  }
+
+  async _regetMetadata(key: string): Promise<AIStudioFileObject> {
+    // Sleep for some time period
+    // eslint-disable-next-line no-promise-executor-return
+    await new Promise(resolve => setTimeout(resolve, this.retryTime));
+
+    // Fetch the latest metadata
+    return this._getMetadata(key);
+  }
+
+  async _set([key, blob]: [string, MediaBlob]): Promise<AIStudioFileSaveResponse> {
+    const response = await super._set([key, blob]) as AIStudioFileSaveResponse;
+
+    // console.log('response.data', response.data);
```

Comment:
Remove log

Reply:
I really wish we had a good way to be able to turn these kinds of logs on, however.